protected/commands/FilesCommand.php

```diff
@@ -21,12 +22,86 @@ class FilesCommand extends CConsoleCommand
         $helpText .= "Usage: ./protected/yiic files checkUrls --doi=<DOI>" . PHP_EOL;
+        $helpText .= "Updates md5 checksum attribute value for all files in a given dataset" . PHP_EOL;
+        $helpText .= "Usage: ./protected/yiic files updateMD5FileAttributes --doi=<DOI>" . PHP_EOL;
 
         return $helpText;
     }
 
+    /**
+     * Update MD5 checksum attribute for all files in a dataset given its DOI
+     * 
+     * @param $doi
+     * @return void
+     */
+    public function actionUpdateMD5FileAttributes($doi) {
+        try {
+            $url = $this->findDatasetMd5FileUrl($doi);
+            echo "Processing $url".PHP_EOL;
+
+            // Dataset id is required for querying files
+            $dataset = Dataset::model()->findByAttributes(array(
+                'identifier' => $doi,
+            ));
+
+            # Download and parse dataset md5 file
+            $contents = DownloadService::downloadFile($url);
+            $lines = explode("\n", $contents);
+            foreach ($lines as $line) {
+                $tokens = explode("  ", $line);
+                // Only parse lines with content in md5 file
+                if($tokens[0] !== "") {
+                    $md5_value = $tokens[0];
+                    $filename = basename($tokens[1]);
+                    if ($filename === "$doi.md5")  // Ignore $doi.md5 file
+                        continue;
+
+                    # Update file_attributes table with md5 checksum value
+                    $file = File::model()->findByAttributes(array(
+                        'dataset_id' => $dataset->id,
+                        'name' => $filename,
+                    ));
+                    $file->updateMd5Checksum($md5_value);
+                }
+            }
+        }
+        catch (Exception $e) {
+            echo $e->getMessage();
+        } 
+        catch (\GuzzleHttp\Exception\GuzzleException $ge) {
+            echo $ge->getMessage();
+        }
+    }
+
+    /**
+     * Returns the URL for a dataset's md5 file
+     *
+     * Determines the URL of the md5 file as it could be:
+     * https://ftp.cngb.org/pub/gigadb/pub/10.5524/100001_101000/100006/100006.md5
+     * https://ftp.cngb.org/pub/gigadb/pub/10.5524/101001_102000/101001/101001.md5
+     * https://ftp.cngb.org/pub/gigadb/pub/10.5524/102001_103000/102236/102236.md5
+     *
+     * @param $doi
+     * @return string
+     * @throws Exception
+     */
+    private function findDatasetMd5FileUrl($doi): string
+    {
+        // Directory names representing ranges of dataset DOIs
+        $ranges = ['104001_105000', '103001_104000', '102001_103000', '101001_102000', '100001_101000'];
+
+        foreach ($ranges as $range) {
+            $url = Yii::app()->params['ftp_connection_url']."/pub/gigadb/pub/10.5524/$range/$doi/$doi.md5";
+            // Check URL resolves to a real file
+            $file_exists = @fopen($url, 'r');
```

Comment:
Hi @pli888,

If we are interested only in checking the existence of the md5 file, it's a bit of bandwidth waste to have to download the file to do that check.
May be it's enough to send a HEAD request to the URL endpoint and check the HTTP response is 200 OK?

If so, there is an example in the function beneath it in the same file that uses the PHP's builtin ``get_headers`` to do something identical.

That said, since I talked about standardising on Guzzle, may be we can use Guzzle in both functions (bonus is that it's easier with Guzzle, as we just call ``head()`` on the web client)

Also, I think we should move away from using ``fopen`` for URLs, as that PHP function was initially (and still primarily is) for opening local file streams
and because requests to URLs have their own standardised semantics for handling success and failure that are not well represented in ``fopen``'s file orientated interface.

Reply:
A static function called `fileExists` has now been created in `DownloadService` class. It uses Guzzle `head()` to check a remote file exists. Unit tests in `tests/unit/DownloadTest.php` check the functionality of the functions in `DownloadService`.